### ISSUE TITLE
[Wave Collect] [Xero] Hide guides as exporters 

### ISSUE THREAD

**Issue by rushatgabhane** [Expensify/App#42345]
We should hide the assignedGuide in the preferred exporter list, like we hide them from the members table on the workspace. There should be logic for this somewhere from QBO to follow.

Also, the current preferred exporter should be highlighted in the selection list.

<img width="1065" alt="image" src="https://github.com/Expensify/App/assets/29673073/aef33a4f-b380-4a24-9809-bdd919a465e3">

**Comment by rushatgabhane** [comment]
thank you, i couldn't reproduce it on my account but let me debug it on a new account

**Comment by trjExpensify** [comment]
Any luck on that?

**Comment by rushatgabhane** [comment]
@trjExpensify yes, im able to repro this on a new workspace

<img width="406" alt="image" src="https://github.com/Expensify/App/assets/29673073/9ff23535-2908-4e75-8de7-99d1f86a04cb">

**Comment by rushatgabhane** [comment]
@trjExpensify this same bug is repro for Quickbooks too -

This is because `exportConfiguration.exporter` is `undefined` on connect and remains undefined till we manually select.
I think this should be fixed on the backend when the connection is made. @lakchote what do you think?

<img width="1048" alt="image" src="https://github.com/Expensify/App/assets/29673073/bf7431f0-71a4-464e-bce4-c0d41133d8e4">

**Comment by trjExpensify** [comment]
Looking at this now.

**Comment by trjExpensify** [comment]
This seems kinda' strange, why can't I find `exportConfiguration.exporter` in the policyConnections data in the console on OldDot? 

`g_policyPage.policy.policy.connections.xero.config`

<img width="795" alt="image" src="https://github.com/Expensify/App/assets/16232057/816d4632-6d46-401c-99a0-f3714aa1fa88">

**Comment by francoisl** [comment]
Is it possible you never opened the config modal in OldDot and clicked "Save"? It's supposed to open by default when you connect the policy, and you're supposed to choose which accounts to export to, etc. My guess is that the exporter is only saved at that point.

**Comment by trjExpensify** [comment]
>Is it possible you never opened the config modal in OldDot and clicked "Save"? 

Yeah, I made the connection via NewDot. But I would have thought at least the preferredExporter would be there and null if not set? I can try again afresh on OldDot.

**Comment by trjExpensify** [comment]
Okay yeah.. so a couple of things: 

1. After making the connection and the config settings pop-up opens, the preferred exporter has a value selected, but the `exporter` field does not appear in the policyConnections data
2. I hit `Save` and then the `exporter` field appears in the policyConnections data. 

https://github.com/Expensify/App/assets/16232057/9187047c-3d68-4aab-bad0-b2467d50a566

**Comment by trjExpensify** [comment]
Not sure how that is going to translate to NewDot though. We don't have to click "Save" after making the connection to trigger that. 

>@trjExpensify this same bug is repro for Quickbooks too -

By contrast, it seems fine and selected when I connect a new workspace to QBO on NewDot: 

<img width="1504" alt="image" src="https://github.com/Expensify/App/assets/16232057/36c48064-08db-438b-9d8d-b0a14eb257e8">

**Comment by francoisl** [comment]
> After making the connection and the config settings pop-up opens, the preferred exporter has a value selected, but the exporter field does not appear in the policyConnections data

It's "selected" because it's the first option in the dropdown and that's what your browser does if we don't specify which value it should show as selected by default, but it's not saved in the policy config.
We need a small backend change to set a default value for the exporter when we create the config, feel free to open an internal issue and assign it to me.

**Comment by francoisl** [comment]
(Or I guess we can reuse this one too 😄)

**Comment by trjExpensify** [comment]
Haha yeah, we can just use this one. I'll co-assign yah.. thanks!

**Comment by francoisl** [comment]
The backend changes to always set an `exporter` are on staging now.

**Comment by trjExpensify** [comment]
@rushatgabhane @hungvu193 were paid centrally for the Xero project, closing!
